### PR TITLE
Idle timeout

### PR DESCRIPTION
**Pull request by spirali**

### CODE REVIEW

File: src/bin/hq.rs [hunk 1]
```diff
@@ -70,6 +71,9 @@ struct ServerStartOpts {
     /// Hostname/IP of the machine under which is visible to others, default: hostname
     #[clap(long)]
     host: Option<String>,
+
+    #[clap(long)]
+    idle_timeout: Option<ArgDuration>,
```
Comment (Kobzol) [inline src/bin/hq.rs:76]: Hmm, I would expect this to be a parameter of the worker, not the server :thinking:

File: tests/test_worker.py [hunk 2]
```diff
@@ -101,3 +100,18 @@ def test_worker_list_resources(hq_env: HqEnv):
...
     assert table[1][3] == "1x10 cpus"
     assert table[2][3] == "4x5 cpus"
+
+
+def test_idle_timeout(hq_env: HqEnv):
+    hq_env.start_server(args=["--idle-timeout", "1s"])
+    w = hq_env.start_worker(args=["--heartbeat", "500ms"])
+    time.sleep(0.5)
+    hq_env.command(["submit", "--", "sleep", "1"])
+    time.sleep(1.0)
```
Comment (Kobzol) [inline tests/test_worker.py:110]: `wait_for_job_state`?

**Review by Kobzol — APPROVED** [review]

File: src/bin/hq.rs (hunk 1 above)
Comment (spirali) [reply src/bin/hq.rs:76]: I was thinking about it as kind of "session" parameter where session is a single start of server. But maybe we can have both server setting (that provides a default) and worker setting that may override the default

File: tests/test_worker.py (hunk 2 above)
Comment (spirali) [reply tests/test_worker.py:110]: I am testing here that worker is not turned off too soon. So putting here an wait_for_job state could be placed here, but still some sleep is required

Comment (Kobzol) [reply tests/test_worker.py:110]: Yeah just the `time.sleep(1.0)` should be replaced with `wait_for_job_state(hq_env, 1, "FINISHED")`.

**Pull request merged**